test/functional/api/v2/fact_value_extensions_test.rb

```diff
@@ -0,0 +1,43 @@
+require 'test_helper'
+
+module Api
+  module V2
+    class FactValuesControllerExtensionsTest < ActionController::TestCase
+      setup do
+        User.current = User.find_by_login "admin"
+        @facts       = {
+            "interfaces"        => "lo,eth0",
+            "ipaddress"         => "192.168.100.42",
+            "ipaddress_eth0"    => "192.168.100.42",
+            "macaddress_eth0"   => "AA:BB:CC:DD:EE:FF",
+            "discovery_bootif"  => "AA:BB:CC:DD:EE:FF",
+            "memorysize_mb"     => "42000.42",
+            "discovery_version" => "3.0.0",
+        }
+        FactoryGirl.create(:setting,
+                           :name     => 'discovery_hostname',
+                           :value    => 'discovery_bootif',
+                           :category => 'Setting::Discovered')
+        FactoryGirl.create(:setting,
+                           :name     => 'discovery_prefix',
+                           :value    => 'mac',
+                           :category => 'Setting::Discovered')
+      end
+
+      def setup_set_script_name
+        @controller                 = Api::V2::FactValuesController.new
```

Comment:
I think placing `tests "api/v2/fact_values"` below class (line 6) works too, no need to override this method